polars/polars-io/src/ndjson_core/ndjson.rs

```diff
@@ -261,26 +252,26 @@ impl<'a> CoreJsonReader<'a> {
     }
 }
 
-fn parse_lines<'a>(bytes: &[u8], buffers: &mut PlIndexMap<String, Buffer<'a>>) -> Result<usize> {
-    let mut stream = Deserializer::from_slice(bytes).into_iter::<Value>();
-    for value in stream.by_ref() {
-        let v = value.unwrap_or(Value::Null);
-        match v {
-            Value::Object(value) => {
-                buffers
-                    .iter_mut()
-                    .for_each(|(s, inner)| match value.get(s) {
+fn parse_lines<'a>(bytes: &[u8], buffers: &mut PlIndexMap<String, Buffer<'a>>) -> Result<()> {
+    for line in SplitLines::new(bytes, b'\n') {
+        let mut line = line.to_vec();
```

Comment:
Why do we need to own the line here?

Reply:
the docs on simd_json aren't great but from what I can tell, it looks like you always need an owned slice. 
```rust
// simd_json::to_borrowed_value
pub fn to_value(s: &mut [u8]) -> Result<Value>
```

Reply:
https://github.com/simd-lite/simd-json/issues/242

Reply:
> i think i could do a mem transmute to get around the ownership, but I'm not super comfortable with the unsafe apis.

If the parser mutates the slice, we are not allowed to do that.

We should at least keep an owned `Vec<u8>` buffer around. That way we only pay for the `memcpy` and not for the allocation.

Reply:
> We should at least keep an owned Vec<u8> buffer around. 
are there any examples of this in the repo?

Reply:
In pseudo rust code you do this:

```rust
fn parse_impl<'a>(bytes: &[u8], buffers: &mut PlIndexMap<String, Buffer<'a>>, line: &mut Vec<u8>) -> Result<usize> {
    // here we reuse the buffer every iteration
    line.clear();
    line.extend_from_slice(bytes);

    match line.len() {
        0 => Ok(0),
        1 => {
            if line[0] == NEWLINE {
                Ok(1)
            } else {
                Err(PolarsError::ComputeError(
                    "Invalid JSON: unexpected end of file".into(),
                ))
            }

// other code
...
  let mut buf = vec![];
  for line in SplitLines::new(bytes, NEWLINE) {
        offset += 1; // the newline
        offset += parse_impl(line, buffers, &mut buf)?;
    }

// other code

```